### PR TITLE
refactor(theme.js): improve styling so that all the styles come from theme.js

### PR DESCRIPTION
**Pull request by navin-moorthy**
Improve styling so that all the styles come from theme.js

### CODE REVIEW

**Comment by navin-moorthy** [comment]
@prasanna1211 I have differentiated the object into checkbox because when we move to different component, it will be easy to change the theme.js for different projects.

If this will not work, suggest a code review on how the theme object should like and I will change it accordingly

File: app/components/Checkbox/index.js [hunk 1]
```diff
@@ -46,26 +46,23 @@ const Checkbox = ({
 
     {/* This is the control box with a check icon as children */}
     <ControlBox
-      borderWidth="1.5px" // TODO: should come from theme
-      size="14px" // TODO: should come from theme
-      borderColor={
-        isDisabled ? 'checkbox.disabled.borderColor' : 'checkbox.borderColor' // TODO: Based on theme.borderColor
-      }
+      size={0.875}
+      borderWidth={1.5}
```
Comment (prasanna1211) [inline app/components/Checkbox/index.js]: Let width come from theme too. Border width generally has definite set of values.

File: app/components/Checkbox/index.js [hunk 2]
```diff
@@ -46,26 +46,23 @@ const Checkbox = ({
...
-          ? {}
-          : { bg: 'brand.500', color: 'white', borderColor: 'brand.500' } // TODO: Theme.disabled should have the styles
-      }
+      _checked={{
+        bg: 'bg.checkbox.checked',
+        color: 'white',
+        borderColor: 'borderColor.checkbox.checked',
+      }}
+      _disabled={{ borderColor: 'borderColor.checkbox.disabled' }}
+      _checkedAndDisabled={{}}
```
Comment (prasanna1211) [inline app/components/Checkbox/index.js]: this is for ?

File: app/components/Checkbox/index.js [hunk 3]
```diff
@@ -46,26 +46,23 @@ const Checkbox = ({
...
-              borderColor: 'brand.500', // Theme.border
-              shadow: '0 0 0 2px #31a7d840, inset 0 0 0 2px #31a7d840',
-            }
-      }
+      _hover={{
+        borderColor: 'borderColor.checkbox.hover',
+        shadow: 'checkbox.hover',
+      }}
+      _checkedAndHover={{}}
       {...getAllowedProps(override.controlBox)}
```
Comment (prasanna1211) [inline app/components/Checkbox/index.js]: Govind is working on filtering props right ?

Comment (govindsingh55) [reply app/components/Checkbox/index.js]: yes

File: app/components/Checkbox/index.js (hunk 2 above)
Comment (navin-moorthy) [reply app/components/Checkbox/index.js]: image

Above is the checked and disabled state, I thought we can handle this state later.

File: app/components/Checkbox/index.js (hunk 1 above)
Comment (navin-moorthy) [reply app/components/Checkbox/index.js]: Thanks for pointing out, I must have missed it with a confusion.

File: app/components/Checkbox/index.js (hunk 3 above)
Comment (navin-moorthy) [reply app/components/Checkbox/index.js]: Yeah, he is working on filtering the props, these props are for our component styling.

**Pull request merged**